color.go

```diff
@@ -36,7 +36,7 @@ var (
 // Color defines a custom color object which is defined by SGR parameters.
 type Color struct {
 	params  []Attribute
-	noColor *bool
```

Comment:
Why change from using a pointer variable though?

Reply:
The original implementation used a pointer so that it could use `nil` as a `undefined` state. But we don't have to do that to implement a default value because we can just set the default value when the struct's created.
Using a value for fields like `bool` is usually faster than using a pointer.

Reply:
Faster... Hmmmm is it really faster?
In this context using either pointers or regular values will have no effect on the complexity of the program.

You make a valid point with having to work with nil as the default value of the pointer bool but it has to be considered a necessary evil.

You'd notice that `NoColor` is exported and your PR would break consumer code that depends on setting the `NoColor` variable to update the `noColor` struct field by reference. If you stop using a pointer then changing the value of `NoColor` does not automatically change the value of `noColor`

Reply:
That's a valid point. I didn't notice that the default value is a variable.

Reply:
I've reverted related change.